ros/src/pybullet_ros/plugins/control.py

```diff
@@ -76,27 +74,22 @@ def execute(self):
         """this function gets called from pybullet ros main update loop"""
         """check if user has commanded a joint and forward the request to pybullet"""
         # flag to indicate there are pending position control tasks
-        position_ctrl_task = False
-        velocity_ctrl_task = False
-        effort_ctrl_task = False
+        control_params = {"bodyUniqueId": self.robot, "jointIndices": self.joint_indices}
+        command_available = False
         if self.pc_subscriber.get_is_data_available():
-            self.position_joint_commands = self.pc_subscriber.get_last_cmd()
-            position_ctrl_task = True
+            control_params["controlMode"] = self.pb.POSITION_CONTROL
+            control_params["targetPositions"] = self.pc_subscriber.get_last_cmd()
+            control_params["forces"] = self.force_commands
+            command_available = True
         if self.vc_subscriber.get_is_data_available():
-            self.velocity_joint_commands = self.vc_subscriber.get_last_cmd()
-            velocity_ctrl_task = True
+            control_params["controlMode"] = self.pb.VELOCITY_CONTROL
+            control_params["targetVelocities"] = self.vc_subscriber.get_last_cmd()
+            control_params["forces"] = self.force_commands
+            command_available = True
         if self.ec_subscriber.get_is_data_available():
-            self.effort_joint_commands = self.ec_subscriber.get_last_cmd()
-            effort_ctrl_task = True
-        # forward commands to pybullet, give priority to position control cmds, then vel, at last effort
-        if position_ctrl_task:
-            self.pb.setJointMotorControlArray(bodyUniqueId=self.robot, jointIndices=self.joint_indices,
-                                              controlMode=self.pb.POSITION_CONTROL,
-                                              targetPositions=self.position_joint_commands, forces=self.force_commands)
-        elif velocity_ctrl_task:
-            self.pb.setJointMotorControlArray(bodyUniqueId=self.robot, jointIndices=self.joint_indices,
-                                              controlMode=self.pb.VELOCITY_CONTROL,
-                                              targetVelocities=self.velocity_joint_commands, forces=self.force_commands)
-        elif effort_ctrl_task:
-            self.pb.setJointMotorControlArray(bodyUniqueId=self.robot, jointIndices=self.joint_indices,
-                                              controlMode=self.pb.TORQUE_CONTROL, forces=self.effort_joint_commands)
+            control_params["controlMode"] = self.pb.TORQUE_CONTROL
+            control_params["forces"] = self.ec_subscriber.get_last_cmd()
+            command_available = True
+
+        if command_available:
```

Comment:
good catch adding the check in case no commands were sent. However, you might as well just do `if "controlMode" in control_params:` and save the extra variable.